HIPs/solang.md

```diff
@@ -0,0 +1,260 @@
+with the contract on the blockchain.
+
+The compiler has the following stages:
+
+## Parser
+
+The parser using an LR(1) parser generator and a hand-written lexer. This
+component has been spun out into its own rust crate
+[Solang-parser](https://crates.io/crates/Solang-parser).
+The parser has been tested against a huge corpus of Solidity test contracts,
+including the ethereum solidity's own testsuite, which runs on Solang CI.
+
+This crate is being
+used by Foundry's [forge fmt](https://github.com/foundry-rs/foundry/tree/master/fmt) command for a rust-based Solidity Code formatter.
+
+## SEMA
+
+This is the semantic analysis stage. The parse tree is validated, and diagnostics
+(errors and warnings) are generated for invalid syntax. The output of this
+stage is a fully decorated Abstract Syntax Tree (AST) of the source code.
+
+The sema stage is the essential piece of the Solang language server, which powers the [Visual Studio Code Solang Solidity Compiler extension](https://marketplace.visualstudio.com/items?itemName=Solang.Solang). This extension gives squiggly lines
+for errors and warnings, and information for tokens (e.g. variables) when hovering over them.  This was developed under the
+first [Hyperledger Mentorship](https://wiki.hyperledger.org/display/INTERN/Create+a+new+Solidity+Language+Server+%28SLS%29+using+Solang+Compiler).
+
+## Code generation
+
+This stage transforms the AST into a control flow graph (CFG). Solang uses its
+own intermediate representation of the code in CFG form. The advantage is that we can do flow analysis passes on a higher level than LLVM
+can do. For example, accessing state on a blockchain is expensive and we
+remove redundant state stores or state loads from the generated contract code. Here is
+an [overview of all the passes Solang has](https://Solang.readthedocs.io/en/latest/optimizer.html).
+
+The [2nd hyperledger mentorship](https://wiki.hyperledger.org/display/INTERN/Implement+two+compiler+passes+for+the+Solang+Solidity+Compiler) implemented two additional
+passes for improved code generation.
+
+## Emit
+
+The last stage transforms our own intermediate CFG from into LLVM IR. This stage
+is fairly straightfoward.
+
+Solang is licensed under the Apache-2.0 license.
+
+## LLVM Backend
+
+The LLVM backend is used for a selection of optimization passes, and generates
+the correct code for the target chain to create an in-memory object file. This
+is then linked using the LLVM linker into the final file.
+
+## Testing
+
+There is a mock implementation of the Solana, Substrate, ewasm, that serve to
+implement runtime tests and assert the correctness of the compiler's output.
+There are also integration tests which use the actual chain for end-to-end testing:
+compile solidity, deploy it to a node of the blockchain (running in a container),
+and call various contract functions.
+
+# Effort and Resources
+
+Initial funding for Solang provided was through two grants from the Web3
```

Comment:
It would be valuable here if you are able to share some details on the decision process to land this project in Hyperledger, given its lineage and funding from the "native web3" world. I personally would love to see more projects natively born in the web3 world to be hosted in Hyperledger.

Reply:
Good idea. I've added some words